package NYT.NRpc.NProto;

import "yt/ytlib/ytree/attributes.proto";
import "yt/ytlib/misc/guid.proto";
import "yt/ytlib/misc/error.proto";

////////////////////////////////////////////////////////////////////////////////

message TRequestHeader
{
    optional NYT.NProto.TGuid request_id = 1;
    required string path = 2;
    required string verb = 3;
    optional bool one_way = 4;
    optional NYT.NYTree.NProto.TAttributes attributes = 5;
}

message TResponseHeader
{
    optional NYT.NProto.TGuid request_id = 1;
    required NYT.NProto.TError error = 2;
    optional NYT.NYTree.NProto.TAttributes attributes = 3;
}

////////////////////////////////////////////////////////////////////////////////
